src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -75,6 +75,14 @@ namespace session {
 
 namespace {
 
+// TODO: asyncRpc - save_document_diff will call isWebsiteProject() that
+// needs to check the R runtime for whether Rmarkdown is installed. This is
+// not entirely accurate since if the first call to isWebsiteProject() is made
+// offline, it will return the wrong thing. Can we just cache a flag globally
+// whether RMarkdown is available - compute at startup and update when new libs
```

Comment:
Cache it in onDeferredInit

Reply:
The rmarkdownAvailable() method is getting called when initializing some other module, before the onDeferredInit handler so I just am caching it the first time it's used. Even if that module goes away, there's a call to it in SessionRMarkdown's onDeferredInit() so it will definitely be called.